SimGeneral/TrackingAnalysis/plugins/SimHitTPAssociationProducer.cc

```diff
@@ -14,6 +14,17 @@
 
 #include "SimGeneral/TrackingAnalysis/interface/SimHitTPAssociationProducer.h"
 
+namespace {
+  using TkId = std::pair<uint32_t, EncodedEventId>;
+  struct TkIdHash {
+    std::size_t operator()(TkId const &s) const noexcept {
+      std::size_t h1 = std::hash<uint32_t>{}(s.first);
+      std::size_t h2 = std::hash<uint32_t>{}(s.second.rawId());
+      return h1 ^ (h2 << 1);
```

Comment:
We have also https://github.com/cms-sw/cmssw/blob/master/FWCore/Utilities/interface/hash_combine.h with which this could be replaced with 
```suggestion
      return edm::hash_value(s.first, s.second.rawId());
```

Reply:
will verify timing. simpler hashing was casuing too manu collisions

Reply:
Any suggestion for the name of the alias of the pair and for the hash struct (and the file itself)?
I hope that alias and struct can be located in the same file....